### ISSUE TITLE
Does not compile against PostgreSQL 13.1

### ISSUE THREAD

**Issue by grawlinson** [timescale/timescaledb#2774]
**Relevant system information:**
 - OS: Arch Linux  `Linux $HOSTNAME 5.10.2-arch1-1 #1 SMP PREEMPT Mon, 21 Dec 2020 19:59:42 +0000 x86_64 GNU/Linux`
 - PostgreSQL version (output of `postgres --version`): 13.1
 - TimescaleDB version (output of `\dx` in `psql`): 2.0.0
 - Installation method: source

**Describe the bug**
TimescaleDB 2.0.0 fails to compile against PostgreSQL 13.1 due to a changed function signature.

**To Reproduce**
1. Install dependencies
2. Download 2.0.0.tar.gz from GitHub
3. Attempt compilation [FAIL]
4. Patch source with [this](https://github.com/timescale/timescaledb/commit/21dc9b9c1a307e46eda5fa621488ebeb6ee9636c) commit to bypass previous error

**Expected behavior**
Successful compile.

**Actual behavior**

```
-- The C compiler identification is GNU 10.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- TimescaleDB version 2.0.0. Can be updated from version 2.0.0-rc4
-- Build type is Release
-- Install method is 'source'
-- Performing Test CC_SUPPORTS_NO_FORMAT_TRUNCATION
-- Performing Test CC_SUPPORTS_NO_FORMAT_TRUNCATION - Success
-- Performing Test CC_SUPPORTS_IMPLICIT_FALLTHROUGH
-- Performing Test CC_SUPPORTS_IMPLICIT_FALLTHROUGH - Success
-- Performing Test CC_SUPPORTS_VISIBILITY_HIDDEN
-- Performing Test CC_SUPPORTS_VISIBILITY_HIDDEN - Success
-- Using compiler GNU
-- Found Git: /usr/bin/git (found version "2.29.2")
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
-- Unable to get git commit information
-- Using pg_config /usr/bin/pg_config
-- Compiling against PostgreSQL version 13.1
CMake Warning at CMakeLists.txt:314 (message):
  clang-format version 7 or 8 required


CMake Warning at CMakeLists.txt:323 (message):
  clang-format is disabled (can't find clang-format or docker)


-- Assertion checks are OFF
-- Found OpenSSL: /usr/lib/libcrypto.so (found version "1.1.1i")
-- OPENSSL_LIBRARIES: /usr/lib/libssl.so;/usr/lib/libcrypto.so
-- Using OpenSSL version 1.1.1i
-- Using nm /usr/bin/nm
-- Using pg_regress /usr/lib/postgresql/pgxs/src/test/regress/pg_regress
-- Skipping isolation regress checks since program 'pg_isolation_regress' was not found
-- Configuring done
-- Generating done
-- Build files have been written to: /build/timescaledb/src/timescaledb-2.0.0/build
TimescaleDB build system initialized in ./build. To compile, do:
cd ./build && make
Scanning dependencies of target sqlupdatescripts
[  1%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/updates/post-update.sql.processed
[  2%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--2.0.0-rc4--2.0.0.sql
[  2%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--2.0.0-rc3--2.0.0.sql
[  3%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--2.0.0-rc2--2.0.0.sql
[  3%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--2.0.0-rc1--2.0.0.sql
[  4%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.7.4--2.0.0.sql
[  4%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.7.3--2.0.0.sql
[  5%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.7.2--2.0.0.sql
[  5%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.7.1--2.0.0.sql
[  6%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.7.0--2.0.0.sql
[  6%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.6.1--2.0.0.sql
[  7%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.6.0--2.0.0.sql
[  7%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.5.1--2.0.0.sql
[  8%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.5.0--2.0.0.sql
[  8%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.4.2--2.0.0.sql
[  9%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.4.1--2.0.0.sql
[  9%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.4.0--2.0.0.sql
[ 10%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.3.2--2.0.0.sql
[ 10%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.3.1--2.0.0.sql
[ 11%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.3.0--2.0.0.sql
[ 11%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.2.2--2.0.0.sql
[ 12%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.2.1--2.0.0.sql
[ 12%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.2.0--2.0.0.sql
[ 13%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.1.1--2.0.0.sql
[ 13%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--1.1.0--2.0.0.sql
[ 13%] Built target sqlupdatescripts
Scanning dependencies of target sqlfile
[ 13%] Generating /build/timescaledb/src/timescaledb-2.0.0/build/sql/timescaledb--2.0.0.sql
[ 13%] Built target sqlfile
Scanning dependencies of target timescaledb
[ 13%] Building C object src/CMakeFiles/timescaledb.dir/agg_bookend.c.o
[ 14%] Building C object src/CMakeFiles/timescaledb.dir/func_cache.c.o
[ 14%] Building C object src/CMakeFiles/timescaledb.dir/cache.c.o
/build/timescaledb/src/timescaledb-2.0.0/src/cache.c: In function ‘remove_pin’:
/build/timescaledb/src/timescaledb-2.0.0/src/cache.c:116:20: error: too many arguments to function ‘list_delete_cell’
  116 |    pinned_caches = list_delete_cell(pinned_caches, lc, prev);
      |                    ^~~~~~~~~~~~~~~~
In file included from /usr/include/postgresql/server/access/xact.h:20,
                 from /build/timescaledb/src/timescaledb-2.0.0/src/cache.c:7:
/usr/include/postgresql/server/nodes/pg_list.h:564:14: note: declared here
  564 | extern List *list_delete_cell(List *list, ListCell *cell);
      |              ^~~~~~~~~~~~~~~~
make[2]: *** [src/CMakeFiles/timescaledb.dir/build.make:108: src/CMakeFiles/timescaledb.dir/cache.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1011: src/CMakeFiles/timescaledb.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```

**Additional context**

pgSQL v13.1 - [list.c](https://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/backend/nodes/list.c;h=80fa8c84e49ae83ca936fdcd1a01fc9188c7d9fa;hb=6daf725a9c66e880fd76d25279ce00710535e030#l767)
timescaleDB v2.0.0 [cache.c](https://github.com/timescale/timescaledb/blob/2.0.0/src/cache.c#L116)

**Comment by Dessa** [comment]
PG13 is not [fully] supported yet. there is a series of PRs which work towards that, but these didn't make it into 2.0.0

this one in particular is addressed by #2738 which is a subset of #2498 (for easing reviewing the changes im assuming)

**Comment by mfreed** [comment]
We are targeting 2.1 to support PG13.
